pkg/controller/pxc/status.go

```diff
@@ -206,37 +208,28 @@ func (r *ReconcilePerconaXtraDBCluster) updateStatus(cr *api.PerconaXtraDBCluste
 	}
 
 	switch {
-	case (cr.Status.PXC.Status == cr.Status.ProxySQL.Status && cr.Spec.ProxySQL != nil && cr.Spec.ProxySQL.Enabled) ||
-		(cr.Status.PXC.Status == cr.Status.HAProxy.Status && cr.Spec.HAProxy != nil && cr.Spec.HAProxy.Enabled):
-		if cr.Status.PXC.Status == api.AppStateReady {
-			clusterCondition = api.ClusterCondition{
-				Status:             api.ConditionTrue,
-				Type:               api.ClusterReady,
-				LastTransitionTime: metav1.NewTime(time.Now()),
-			}
+	case clusterStatus.ProxyReady() && clusterStatus.PXCReady():
+		clusterCondition = api.ClusterCondition{
+			Status:             api.ConditionTrue,
+			Type:               api.ClusterReady,
+			LastTransitionTime: metav1.NewTime(time.Now()),
 		}
 		cr.Status.Status = cr.Status.PXC.Status
-	case (cr.Spec.ProxySQL == nil || !cr.Spec.ProxySQL.Enabled) &&
-		(cr.Spec.HAProxy == nil || !cr.Spec.HAProxy.Enabled) &&
-		cr.Status.PXC.Status == api.AppStateReady:
+	case !clusterStatus.ProxyEnabled() && clusterStatus.PXCReady():
 		clusterCondition = api.ClusterCondition{
 			Status:             api.ConditionTrue,
 			Type:               api.ClusterReady,
 			LastTransitionTime: metav1.NewTime(time.Now()),
 		}
 		cr.Status.Status = cr.Status.PXC.Status
-	case cr.Status.PXC.Status == api.AppStateError ||
-		cr.Status.ProxySQL.Status == api.AppStateError ||
-		cr.Status.HAProxy.Status == api.AppStateError:
+	case clusterStatus.ClusterError():
 		clusterCondition = api.ClusterCondition{
 			Status:             api.ConditionTrue,
 			Type:               api.ClusterError,
 			LastTransitionTime: metav1.NewTime(time.Now()),
 		}
 		cr.Status.Status = api.AppStateError
-	case cr.Status.PXC.Status == api.AppStateInit ||
-		(cr.Spec.ProxySQL != nil && cr.Status.ProxySQL.Status == api.AppStateInit) ||
-		(cr.Spec.HAProxy != nil && cr.Status.HAProxy.Status == api.AppStateInit):
+	case clusterStatus.ClusterInit():
```

Comment:
Here (in [L238](https://github.com/percona/percona-xtradb-cluster-operator/pull/563/files#diff-b246f9cb3cd35db1c2d9931e00d89aabR238)) the AppStateInit being set and then later in [L259](https://github.com/percona/percona-xtradb-cluster-operator/pull/563/files#diff-b246f9cb3cd35db1c2d9931e00d89aabR259) basically a duplication. And for that duplication, we're maintaining "inProgres" variable a doing quite a bit extra stuff. I believe setting the AppStateInit here in sewitch-case is enough